02-peer-protocol.md

```diff
@@ -305,6 +325,9 @@ funding transaction and both versions of the commitment transaction.
     1. type: 0 (`upfront_shutdown_script`)
     2. data:
         * [`...*byte`:`shutdown_scriptpubkey`]
+    1. type: 1 (`channel_type`)
+    2. data:
+        * [`channel_type`:`type`]
```

Comment:
NB: it's a bit wasteful here to reuse `channel_type`, because the length will be specified twice (once because it's a TLV, and then again inside the TLV with the `u16`:`length`).

Reply:
But I guess it prevents us from reusing existing codecs as is, so probably not worth it for a couple bytes...

Reply:
True, but it's pretty neat inside the code, as we're dealing with channel_type everywhere ("do these channel types match?").  This is especially true since we autogenerate our code from the spec, as does lnprototest.